pkg/pipeline/conntrack/conntrack.go

```diff
@@ -42,14 +45,80 @@ type ConnectionTracker interface {
 	Track(flowLogs []config.GenericMap) []config.GenericMap
 }
 
-type conntrackImpl struct {
-	config api.ConnTrack
-	hasher hash.Hash
+//////////////////////////
+
+// TODO: Does connectionStore deserve a file of its own?
+
+// connectionStore provides both retrieving a connection by its hash and iterating connections sorted by their last
+// update time.
+type connectionStore struct {
 	// TODO: should the key of the map be a custom hashStrType instead of string?
-	hash2conn                 map[string]connection
+	hash2conn map[string]*list.Element
+	connList  *list.List
+}
+
+type processConnF func(connection) (shouldDelete, shouldStop bool)
+
+func (ct *connectionStore) addConnection(hashStr string, conn connection) {
+	_, ok := ct.getConnection(hashStr)
+	if ok {
+		log.Errorf("BUG. connection with hash %v already exists in store. %v", hashStr, conn)
+	}
+	e := ct.connList.PushBack(conn)
+	ct.hash2conn[hashStr] = e
+}
+
+func (ct *connectionStore) getConnection(hashStr string) (connection, bool) {
+	elem, ok := ct.hash2conn[hashStr]
+	if ok {
+		conn := elem.Value.(connection)
+		return conn, ok
+	}
+	return nil, ok
+}
+
+func (ct *connectionStore) updateConnectionTime(hashStr string, t time.Time) {
+	elem, ok := ct.hash2conn[hashStr]
+	if !ok {
+		log.Errorf("BUG. connection hash %v doesn't exist", hashStr)
+	}
+	elem.Value.(connection).setLastUpdateTime(t)
+	// move to end of list
+	ct.connList.MoveToBack(elem)
+}
+
+func (ct *connectionStore) iterateOldToNew(f processConnF) {
+	for e := ct.connList.Front(); e != nil; e = e.Next() {
+		conn := e.Value.(connection)
+		shouldDelete, shouldStop := f(conn)
+		if shouldDelete {
+			delete(ct.hash2conn, hex.EncodeToString(conn.getHash().hashTotal))
+			ct.connList.Remove(e)
+		}
+		if shouldStop {
+			break
+		}
+	}
+}
+
+func newConnectionStore() *connectionStore {
+	return &connectionStore{
+		hash2conn: make(map[string]*list.Element),
+		connList:  list.New(),
+	}
+}
+
+//////////////////////////
+
+type conntrackImpl struct {
+	clock                     clock.Clock
+	config                    api.ConnTrack
+	hasher                    hash.Hash
```

Comment:
Some points:

- As you always use `fnv.New32a()` in both production and testing, maybe you don't need to inject it here and just instantiate it directly in `ComputeHash`.
- Instead of `hash.Hash`, I'd use the `hash.Hash32` or `hash.Hash64` interfaces, which also provide a `Sum32() uint32` or `Sum64() uint64` methods. Using them would avoid having to convert byte slices to strings to store/compare the hashes, and we wouldn't need the `areHashEqual` and `copyTotalHash` functions
- Reusing a single hasher instance (using the `Reset()` method) could corrupt the hashes if it is used concurrently. I'd avoid injecting it as a property, but if you still want to inject the hasher for other reasons, I'd change this attribute by:
```go
hasherProvider  func() hash.Hash32
```
(or `hash.Hash64`). This way, when you invoke e.g. `ComputeHash` you can do it like:
```go
 		computedHash, err := ComputeHash(fl, ct.config.KeyDefinition, ct.hasherProvider())
```
So each invocation of `ComputeHash` will use a fresh/unique instance of a hasher.

When you instantiate the `connTrackImpl`, you should set the value as:
```go
hasherProvider: fnv.New32a,
```